### PR TITLE
feat: support latest binary downloading

### PR DESCRIPTION
**Pull request by maiste**
This PR adds a `/latest/<arch>` URL to the Dune Binary Distribution system. Also, it provides a mechanism to update the binaries every time it is build with the latest version. It makes it easier for users to grab the dune version. 

The website displays a `Latest` section at the top, which are just pointers to the latest built binaries.

This PR must be merged after #4 to make sure we have certificates available for the latest build.

### CODE REVIEW

File: sandworm/bin/main.ml [hunk 1]
```diff
@@ -6,11 +6,15 @@ let main commit has_certificate =
...
-  let daily_bundle_date = Metadata.Bundle.get_data_string_from daily_bundle in
+  let daily_bundle_date = Metadata.Bundle.get_date_string_from daily_bundle in
   let s3_daily_bundle = Filename.concat Config.s3_bucket_ref daily_bundle_date in
   let () =
     Rclone.copy ~config_path:Config.rclone_path Config.artifacts_path s3_daily_bundle
   in
+  let latest_bundle = Filename.concat Config.s3_bucket_ref "latest" in
+  let () =
+    Rclone.copy ~config_path:Config.rclone_path Config.artifacts_path latest_bundle
+  in
```
Comment (Leonidas-from-XIV) [inline sandworm/bin/main.ml:17]: I don't know if that is possible on a S3 bucket content leve but I think it would be preferable to me if we did a `30x` redirect on the `/latest` to the latest date instead of changing the content where latest is pointing to.

Comment (Leonidas-from-XIV) [reply sandworm/bin/main.ml:17]: Its not a DNS thing, it's a HTTP thing. It can be [configured on S3](https://docs.aws.amazon.com/AmazonS3/latest/userguide/how-to-page-redirect.html) but not from the bucket contents itself.

Comment (maiste) [reply sandworm/bin/main.ml:17]: Yes, but here we are not using the S3 interface anymore. The first iteration is of the work was using _RClone_ with an S3 bucket. This version is using _RClone_ with SFTP. I kept the S3 wording, as the _RClone_ tends to refer a lot to it in the documentation. 

In terms of architecture, the website is served using GitHub Pages whereas the binaries and certificates are served using Caddy (https://caddyserver.com/docs/quick-starts/static-files). As a result, we don't have something related to S3. The only access we have from our perspective is the file system structure.

TL;DR: unfortunately, we can't do this in the current architecture :smile:

**Comment by Leonidas-from-XIV** [comment]
I am still not a great fan of just having a `latest` URL which will return you a random dune. So how about an alternative idea:

The document which contains the preview links knows the URLs already, so instead of specifying

```
curl -o dune https://download.dune.ci.dev/latest/<arch>/dune
```

it could specify

```
$ curl -o dune <url>
```

likewise all the previews could instead of

```
$ curl https://download.dune.ci.dev/<yyyy-mm-dd>/<arch>/attestation.jsonl -o attestation.jsonl
```

have the actual URLs there. Then the whole verify section can be removed because for each preview and arch the instructions would contain the correct URL and the correct command?

**Comment by maiste** [comment]
This is definitely something we could do to improve the readability on the website. The issue is that we miss the main use case for the _latest_ feature. The goal is to have a URL where you don't have to think about and just pull the latest binary. It is not more random than having the binary from one day to another. 

I understand the concern with the fact that running `curl .../latest/dune` would introduce an issue with tracking the origin of the binary (as it wouldn't be reproducible). However, if your proposal is great and allows the user to just copy and paste the result, it would still have the issue of the single download URL. Maybe this is something we can't do in the current configuration, and we need to check with the Infra team if we want to invest more time to have a finer control on the solution?

**Comment by Leonidas-from-XIV** [comment]
Ok but what is the problem we want to solve? The "latest" URL to be fixed is only a problem if we want e.g. tools to download the latest version, but for people I'd argue being able to copy-paste the actual command is more useful than to have a URL that always contains the most recent binary.

**Comment by maiste** [comment]
In our case, I would say yes, it is easier for the user :smile: 
In the future, to have something more "opam" like (or other programming language like), we should definitely consider to have a better system to solve this (with your suggestion for the `303` redirect on latest).  But it would be a problem for future us, so that's ok!

**Comment by maiste** [comment]
@Leonidas-from-XIV, I've updated following your recommendations. Users can now just copy and paste the daily build :+1:

**Review by Leonidas-from-XIV — APPROVED** [review]
